javascript/code-challenges/challenges-04.test.js

```diff
@@ -0,0 +1,357 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function named updateAnimal that takes in array of animals (strings) and some callback function.
+
+This function should first create a new array. Then iterate over the input array and modify each value based on the callback function provided.
+
+Push each updated animal string into the new array. Return the new array.
+
+HINT: Look at the tests to see how the callback functions are used.
+
+------------------------------------------------------------------------------------------------ */
+
+function upper(str) {
+  return str.toUpperCase();
+}
+
+function lower(str) {
+  return str.toLowerCase();
+}
+
+const updateAnimal = (arr, callback) => {
+  return arr.map(callback);
+};
```

Comment:
nice use of map! I'm actually shocked that works, I would have assumed you'd need to invoke you callback within your map
```js
return arr.map(animal => callback(animal));
```
that said, it's cool that it does work!